### ISSUE TITLE
 option.formattingFn rerender 

### ISSUE THREAD

**Issue by Sweechirly** [glennreyes/react-countup#376]
i  hope to change option.formattingFn to rerender if i changed it's inner params .
```js
const countUpProps = {
 start: start, 
 end: end,
 duration: duration ，
 separator: separator,
 decimals: decimals,
 delay: 0,
 formattingFn: formatNumber // dynamic function
};
```

**Comment by Luc9802** [comment]
@Sweechirly have you had luck to find out any workaround maybe?

**Comment by Sweechirly** [comment]
> @Sweechirly have you had luck to find out any workaround maybe?

unfortunately, no, I made a key for the render element

**Comment by Luc9802** [comment]
> > @Sweechirly have you had luck to find out any workaround maybe?
> 
> unfortunately, no, I made a key for the render element

What do you mean with "I made a key for the render element"?
Have you got an example?

**Comment by Sweechirly** [comment]
> > > @Sweechirly have you had luck to find out any workaround maybe?
> > 
> > 
> > unfortunately, no, I made a key for the render element
> 
> What do you mean with "I made a key for the render element"?
> Have you got an example?

Everytime I refresh my data, the React Component will refresh, and the react-countup will rerender

**Comment by Luc9802** [comment]
> > > > @Sweechirly have you had luck to find out any workaround maybe?
> > > 
> > > 
> > > unfortunately, no, I made a key for the render element
> > 
> > 
> > What do you mean with "I made a key for the render element"?
> > Have you got an example?
> 
> Everytime I refresh my data, the React Component will refresh, and the react-countup will rerender

Do you mean using a key prop directly in CountUp component to let it re-render? If yes, do you know if this is a correct pattern, I knew key prop is usually used only in array map contexts.

**Comment by mmarkelov** [comment]
@Luc9802 could you please provide small representation? i couldn't understand the case

**Comment by Luc9802** [comment]
> @Luc9802 could you please provide small representation? i couldn't understand the case

Here you go @mmarkelov : https://codesandbox.io/s/wonderful-bhabha-do1h0?file=/src/App.js

Basically my app needs to change locale runtime and I'm formatting my value through Intl.NumberFormat.

Please kindly note:
1. uncommenting key prop it actually works as I would expect
2. prefix and suffix are not considered when using formattingFn prop, in my opinion they should be applied like in default scenario or they could arrive as function params for formattingFn

**Comment by mmarkelov** [comment]
@Luc9802 just published **4.4.0** and it should work with **redraw** option:
```js
import "./styles.css";
import CountUp from "react-countup";
import React, { useState } from "react";

const lngs = ["en", "it"];
export default function App() {
  const [lngIdx, setLngIdx] = useState(0);

  const changeLanguage = () => {
    setLngIdx((prevLngIdx) => (prevLngIdx === 0 ? 1 : 0));
  };

  return (
    <div className="App">
      <CountUp
        end={1650.25}
        decimals={2}
        prefix="test"
        suffix="test"
        redraw
        formattingFn={(val) => {
          return new Intl.NumberFormat(lngs[lngIdx]).format(val);
        }}
      />
      <br />
      <button onClick={changeLanguage}>CHANGE LANGUAGE</button>
    </div>
  );
}
```

And i'll investigate the case with passing new `formattingFn` option.

**Comment by Luc9802** [comment]
@mmarkelov I tried 4.4.0 and it works with redraw option.
In your opinion is it possibile to prevent the animation and just re-run the formattingFn when needed somehow? In the example above when the locale changes so.


For what cocerns prefix and suffix not applied in this case should I open another issue?

**Comment by mmarkelov** [comment]
> In your opinion is it possibile to prevent the animation and just re-run the formattingFn when needed somehow?

I'll check it, not sure

> For what cocerns prefix and suffix not applied in this case should I open another issue

Also will check it, but i suppose it's just similar to `CountUp` behaviour.

**Comment by Luc9802** [comment]
@mmarkelov Thank you so much for your effort, appreciated.

**Comment by mmarkelov** [comment]
> For what cocerns prefix and suffix not applied in this case should I open another issue?

Checked this. The same behavior is for original **Countup**, as workaround you can use prefix and suffix in `formattingFn `:
```js
formattingFn={(val) => `${prefix}${new Intl.NumberFormat(lngs[lngIdx]).format(val)}${suffix}`}
```

**Comment by mmarkelov** [comment]
> In your opinion is it possibile to prevent the animation and just re-run the formattingFn when needed somehow?

I suppose that there is no way to do it **Countup** doesn't provide any methods to update options - so we need to create new instance for new options

**Comment by Luc9802** [comment]
@mmarkelov Thank you for the checks.

Sorry to bother you again but should it behave the same way with useCountUp hook for what concerns the formattingFn?
I made another example: https://codesandbox.io/s/strange-cdn-ur0xr?file=/src/App.js

**Comment by mmarkelov** [comment]
@Luc9802 i suppose that there is a bug in `useCountUp`. I'm going to fix it and publish new soon

**Comment by mmarkelov** [comment]
should be fixed in **react-countup@5.0.0-0**
